WebAppServiceBus/WebAppServiceBus/ReceivedMessageStore.cs

```diff
@@ -1,38 +1,53 @@
-﻿using Microsoft.Azure.ServiceBus;
-using Microsoft.Azure.ServiceBus.Primitives;
+﻿using Azure.Messaging.ServiceBus;
 using System.Collections.Generic;
-using System.Text;
+using System;
+using Azure.Identity;
 using System.Threading.Tasks;
 using WebAppServiceBus.Models;
 
 namespace WebAppServiceBus
 {
     public static class ReceivedMessageStore
     {
-        private static QueueClient _receiveClient = null;
         private static List<string> _receivedMessages = new List<string>();
 
-        public static void Initialize(ServiceBusConfiguration config)
+        public static async Task InitializeAsync(ServiceBusConfiguration config)
         {
-            if (_receiveClient != null)
+            var options = new ServiceBusProcessorOptions
             {
-                return;
+                AutoCompleteMessages = false,
+
+                MaxConcurrentCalls = 2
+            };
+
+            ServiceBusClient client = new ServiceBusClient(config.Namespace, new DefaultAzureCredential());
+            ServiceBusProcessor processor = client.CreateProcessor(config.Queue, options);
+
+            processor.ProcessMessageAsync += MessageHandler;
+            processor.ProcessErrorAsync += ErrorHandler;
+
+            async Task MessageHandler(ProcessMessageEventArgs args)
+            {
+                string body = args.Message.Body.ToString();
+                Console.WriteLine(body);
+                _receivedMessages.Add(body);
+
+                // we can evaluate application logic and use that to determine how to settle the message.
+                await args.CompleteMessageAsync(args.Message);
             }
 
-            TokenProvider tokenProvider = TokenProvider.CreateManagedServiceIdentityTokenProvider();
-
-            _receiveClient = new QueueClient($"sb://{config.Namespace}.servicebus.windows.net/", config.Queue, tokenProvider, receiveMode: ReceiveMode.ReceiveAndDelete);
-            _receiveClient.RegisterMessageHandler(
-                (message, cancellationToken) =>
-                {
-                    _receivedMessages.Add($"MessageId:{message.MessageId}, Seq#:{message.SystemProperties.SequenceNumber}, data:{Encoding.UTF8.GetString(message.Body)}");
-                    return Task.CompletedTask;
-                },
-                (exceptionEvent) =>
-                {
-                    _receivedMessages.Add($"Exception: \"{exceptionEvent.Exception.Message}\" {exceptionEvent.ExceptionReceivedContext.EntityPath}");
-                    return Task.CompletedTask;
-                });
+            Task ErrorHandler(ProcessErrorEventArgs args)
+            {
+                // the error source tells me at what point in the processing an error occurred
```

Comment:
Let's match what was in the previous version of the exception handler - https://github.com/Azure-Samples/app-service-msi-servicebus-dotnet/pull/6/files#diff-d7b10eb91c0fa2a720c5fa52dd1bbfcec9959e87e8c028c62bb6a0c9b45da739L33-L34